lib/api.js

```diff
@@ -437,5 +482,23 @@ class LedgerAgent {
       ledgerBlockService: options.ledgerBlockService,
       ledgerQueryService: options.ledgerQueryService
     };
+
+    // FIXME: this should have a more appropriate name to store the plugins
+    // enabled for this ledgerAgent instance.  `plugins`???
+    this.services = options.services;
+
+    // FIXME: there was mention that the various service names offered by the
+    // ledgerAgent should be included in a context. How does this provide for
+    // extensibility of these services?
+
+    // TODO: it may be desirable for plugins to define numerous additional
+    // endpoints. How would those additional endpoints be advertized?
+
+    for(const serviceName of this.services) {
+
+      // FIXME: should plugins be namespaced like this to avoid collisions?
+      this.service[`${serviceName}Service`] =
```

Comment:
-1 to this approach. Let's require the plugin to specify its name and if it isn't unique, we disallow it. We should also be clear about the naming conventions ... especially since it looks like this stuff is going into URLs?